ui/src/main/js/page/user/user/UserEditModal.js

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { saveUser, validateUser } from 'store/actions/users';
+import Modal from 'common/component/modal/Modal';
+import DynamicSelectInput from 'common/component/input/DynamicSelectInput';
+import PasswordInput from 'common/component/input/PasswordInput';
+import TextInput from 'common/component/input/TextInput';
+import * as HTTPErrorUtils from 'common/util/httpErrorUtilities';
+import UserEnum from 'common/util/enums/User';
+
+const UserEditModal = ({ data, isOpen, toggleModal }) => {
+    const dispatch = useDispatch();
+    const { external } = data;
+    const [userData, setUserData] = useState(data);
+    
+    const fieldErrors = useSelector(state => state.users.error.fieldErrors);
+    const inProgress = useSelector(state => state.users.inProgress);
+    const roles = useSelector(state => state.roles.data);
+    const saveStatus = useSelector(state => state.users.saveStatus);
+
+    useEffect(() => {
+        if ( saveStatus === 'VALIDATED' && !inProgress) { 
+            handleSave();
+        }
+    }, [saveStatus, inProgress]);
+
+    function handleClose() {
+        toggleModal(false);
+    }
+
+    const handleOnChange = (label) => {
+        return ({ target: { value } }) => {
+            setUserData(userData => ({...userData, [label]: value }));
+        }
+    }
+    
+    function handleSave() {
+        dispatch(saveUser(userData));
+        handleClose();
+    }
+
+    function handleSubmit() {
+        if (passwordsMatch(userData)) {
```

Comment:
move `passwordsMatch` function to above this function.  It is being called before it is defined.